Overhaul CI documentation (#183)
freedomofpress/securedrop-dev-docs: docs/testing_configuration_tests.rst

the Ansible layout, the tests are rather tightly coupled to hosts. The
layout of config tests is therefore subject to change.

Running the CI Staging Environment
----------------------------------

> The staging environment can also run via CI, running in GCE. These tests are

Reviewer: GCE == Google Cloud Environment? This might be worth linking to or expanding the acronym.

Author: I thought it was "Google Compute Engine" but yeah, let me expand it first.

> The staging environment can also run via CI, running in GCE. These tests are

Reviewer: ```suggestion
The staging environment can also run via CI in GCE. These tests are
```

> run every night or if you push to a branch that starts with ``stg-``. Currently
> this can only be done by members of the ``freedomofpress`` GitHub organization,
> please ask if you'd like someone to run the tests for you.

Reviewer: ```suggestion
run every night or if a member of the ``freedomofpress`` Github Organization 
pushes to a branch that starts with ``stg-``. Please ask if you'd like someone to run the tests for you.
```

> please ask if you'd like someone to run the tests for you.

Reviewer: Is it well understood where people should ask for this? Is it worth including an email/link to the github issues page where people should submit this request?

Author: I'll say "Please ask in your PR if..."

> You can also run them yourself if you have a Google Cloud Platform account and Docker
> installed locally.

Reviewer: ```suggestion
The tests can also be run manually with a Google Cloud Platform account and Docker
installed locally:
```

Source the setup script using the following command:

.. code:: sh

    source ./devops/gce-nested/ci-env.sh

You will be prompted for the values of the required environment variables. There
are some defaults set that you may want to change. You will need to export
``GOOGLE_CREDENTIALS`` with authentication details for your GCP account,

> which is outside the scope of this guide. Some parameters are specific to FPF's

Reviewer: Is it worth linking to a google auth setup tutorial, or even their general auth documentation [here](https://cloud.google.com/docs/authentication/use-cases)? I'm not sure I'd know how to get started with this without digging into the code to see what tools it's using and how it's using them.

Author: Yeah, I can link to that page, but tbh I wouldn't know how to provision the credential either.

Reviewer: Good point. Google seems to change their docs and auth schemas relatively often too, so even if we did put something here it might change in the next few months.

> GCE setup and may need adjusting if you are running elsewhere.

Reviewer: Is it worth enumerating these specific parameters so users can find where to start debugging if they run into problems?

Author: It's kind of a cop-out because I actually have no idea what it would take to run elsewhere. For example, we have a custom built Debian image we use (https://github.com/freedomofpress/securedrop/blob/dbdccc593e1aeab149dbd6019709394ec1e6dc9c/devops/gce-nested/gce-start.sh#L28) that I don't know what it would take to replicate.
